### PR TITLE
Fix changes from 0.9.1

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

**Comment by nishantmonu51** [comment]
👍

File: src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala [hunk 1]
```diff
@@ -19,23 +19,22 @@
 
 package io.druid.indexer.spark
 
+import com.google.common.base.Strings
```
Comment (xvrl) [inline src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala]: import order seems messed up

File: src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala [hunk 2]
```diff
@@ -114,13 +113,14 @@ class TestSparkDruidIndexer extends FlatSpec with Matchers {
...
-              dataSchema.getParser.getParseSpec.getDimensionsSpec.getDimensions.asScala.toSet --
+              dataSchema.getParser.getParseSpec.getDimensionsSpec.getDimensionNames.asScala.toSet --
                 dataSchema.getParser.getParseSpec.getDimensionsSpec.getDimensionExclusions.asScala.toSet
             )
           for (dimension <- qindex.getDimensionNames.iterator().asScala) {
-            val dimVal = qindex.getDimValueLookup(dimension).asScala
+            val dimVal = qindex.getDimValueLookup(dimension).asScala.map(Strings.nullToEmpty)
             dimVal should not be 'Empty
             for (dv <- dimVal) {
+              dv should not be null
```
Comment (xvrl) [inline src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala]: what's the purpose of this check? The index might return null and that's fine, why map it to empty string?

File: src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala [hunk 3]
```diff
@@ -114,13 +113,14 @@ class TestSparkDruidIndexer extends FlatSpec with Matchers {
...
           qindex.getDimensionNames.asScala.toSet should
             equal(
-              dataSchema.getParser.getParseSpec.getDimensionsSpec.getDimensions.asScala.toSet --
+              dataSchema.getParser.getParseSpec.getDimensionsSpec.getDimensionNames.asScala.toSet --
                 dataSchema.getParser.getParseSpec.getDimensionsSpec.getDimensionExclusions.asScala.toSet
             )
           for (dimension <- qindex.getDimensionNames.iterator().asScala) {
-            val dimVal = qindex.getDimValueLookup(dimension).asScala
+            val dimVal = qindex.getDimValueLookup(dimension).asScala.map(Strings.nullToEmpty)
             dimVal should not be 'Empty
```
Comment (xvrl) [inline src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala]: checking toString implementations seems a little brittle.

File: build.sbt [hunk 4]
```diff
@@ -28,11 +28,11 @@ scalaVersion := "2.10.5"
 crossScalaVersions := Seq("2.10.5", "2.11.7")
 
 // Requires 0.8.2 or later and https://github.com/druid-io/druid/pull/1940
-val druid_version = "0.9.0-rc3-SNAPSHOT"
+val druid_version = "0.9.1-SNAPSHOT"
 // This is just used here for Path, so anything that doesn't break spark should be fine
 val hadoop_version = "2.4.0"
 // Requires a patch for https://issues.apache.org/jira/browse/SPARK-11016
-val spark_version = "1.5.2-mmx1"
+val spark_version = "1.5.2-mmx4"
```
Comment (xvrl) [inline build.sbt]: anything new here we need?

File: src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala (hunk 2 above)
Comment (drcrallen) [inline src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala]: Was encountering an error with null, changed this around to make it more obvious.

File: build.sbt (hunk 4 above)
Comment (drcrallen) [inline build.sbt]: it is scoped `provided` so it shouldn't have anything in particular. The actual version is pulled during pull-deps

Comment (xvrl) [inline build.sbt]: I was just wondering what the difference betweetn mmx1 and mmx4 was

File: src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala (hunk 2 above)
Comment (xvrl) [inline src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala]: what kind of error? is it a bug or expected behavior?

File: build.sbt (hunk 4 above)
Comment (drcrallen) [inline build.sbt]: two small patches https://github.com/metamx/spark/commit/9c95489e900666e4732453f580c905bf97652de4 and https://github.com/metamx/spark/commit/0efd39d2d3ae458d3279719fef2db90dc2b3f882

and the pom is published with hadoop 2.4 as a dependency.

Comment (drcrallen) [inline build.sbt]: I screwed up mmx3, and mmx2 had the wrong hadoop version

File: src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala (hunk 2 above)
Comment (drcrallen) [inline src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala]: change in null behavior. Prior behavior had no null value in the iterable over strings, but current version has a null value in the dimension value iterable. I thiiiink it is expected

File: src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala (hunk 1 above)
Comment (drcrallen) [inline src/test/scala/io/druid/indexer/spark/TestSparkDruidIndexer.scala]: re-ran formatting

File: version.sbt [hunk 5]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.9.0-1-SNAPSHOT"
\ No newline at end of file
+version in ThisBuild := "0.9.1-0-SNAPSHOT"
```
Comment (xvrl) [inline version.sbt]: missing end of line

Comment (drcrallen) [inline version.sbt]: fixed

**Pull request merged**